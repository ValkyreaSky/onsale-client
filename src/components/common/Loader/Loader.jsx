import React from 'react';
import styled from 'styled-components';
// Components
import { Spin } from 'antd';

const Container = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  height: 20rem; 
`;

const Loader = () => (
  <Container>
    <Spin size="large" />
  </Container>
);

export default Loader;
