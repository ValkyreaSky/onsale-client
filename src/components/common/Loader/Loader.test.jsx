import React from 'react';
import { shallow } from 'enzyme';
import { Spin } from 'antd';
import Loader from './Loader';

describe('<Loader />', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<Loader />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render <Spin> comopnent', () => {
    const wrapper = shallow(<Loader />);
    expect(wrapper.find(Spin).length).toBe(1);
  });
});
