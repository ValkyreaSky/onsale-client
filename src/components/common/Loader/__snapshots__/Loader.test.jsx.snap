// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> should match snapshot 1`] = `
<styled.div>
  <Spin
    prefixCls="ant-spin"
    size="large"
    spinning={true}
    wrapperClassName=""
  />
</styled.div>
`;
